agent-engines/gpu_worker/deployment_pipeline.py

```
@@ -0,0 +1,264 @@
+        logger.info(f"Rolled back deployment {deployment_id}")
+
+    async def _execute_hooks(self, hooks: List[Callable], hook_type: str) -> None:
+        """Execute pre or post deployment hooks."""
+        for hook in hooks:
+            try:
+                if asyncio.iscoroutinefunction(hook):
+                    await hook()
+                else:
+                    hook()
+                logger.debug(f"Executed {hook_type} hook")
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)